src/state/AppContext.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { FC } from "react"
+import { createContext } from "react"
+import { AppSettings } from "./AppSettings"
+
+export interface AppContextProps {
+    settings: AppSettings
+    setSettings: React.Dispatch<React.SetStateAction<AppSettings>>
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const AppContext = createContext<AppContextProps>({
+    settings: new AppSettings({ theme: "predefined" }),
+    setSettings: function (): void {
+        throw new Error("Debug")
+    },
+})
+
+export interface AppStateProviderProps {
+    children: React.ReactNode
+    settings: AppSettings
+    setSettings: React.Dispatch<React.SetStateAction<AppSettings>>
+}
+/**
+ * App state provider component. It will pass down to all its
+ * descendants a settings object {@link AppSettings}
+ */
+export const AppStateProvider: FC<AppStateProviderProps> = props => {
+    return (
+        <AppContext.Provider
+            value={{ settings: props.settings, setSettings: props.setSettings }}
+        >
+            {props.children}
+        </AppContext.Provider>
+    )
+}
```

Comment:
L'intera cartella `state` non mi sembra molto sensata, vero è lo state generale dell'applicazione quello delle impostazioni, ma penso sia meglio dividere i context per scopo dando nomi più descrittivi, banalmente SettingsContext che possa essere consumato per avere le impostazioni, puoi mettere tutta sta roba in un unico file `settings.ts` in `utils` secondo me, no?
praticamente me lo immagino simile al login, che per il momento è fatto così https://github.com/PoliNetworkOrg/PoliFemo/blob/feature/CU-34ga4dy/src/utils/login.ts

Reply:
Okay si ha molto più senso.